app/models/repository.rb

```diff
@@ -29,12 +29,17 @@
 
 class Repository < ActiveRecord::Base
   include Redmine::Ciphering
+  include OpenProject::Scm::ManageableRepository
 
   belongs_to :project
   has_many :changesets, order: "#{Changeset.table_name}.committed_on DESC, #{Changeset.table_name}.id DESC"
 
   before_save :sanitize_urls
 
+  # Managed repository lifetime
+  after_save :create_managed_repository, if: Proc.new { |repo| repo.managed? }
+  after_destroy :delete_managed_repository, if: Proc.new { |repo| repo.managed? }
```

Comment:
I guess that service would orchestrate the two existing repo delayed jobs (CreateManaged.., DeleteManaged) in such a way that they can be seen as the up and down migration of a managed repository.

Then the repositories controller could call some `status` method on the service to check whether the last operation was successful to at least notify the users on the repositories settings page, or the repository page itself.

Does this make any sense?